libutils/sequence.c

```diff
@@ -449,3 +450,87 @@ int SeqStringLength(Seq *seq)
 
     return total_length;
 }
+
+// TODO: These static helper functions could be (re)moved
+static bool HasNulByte(const char *str, size_t n)
+{
+    for (int i = 0; i<n; ++i)
+    {
+        if (str[i] == '\0')
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+static long GetLengthPrefix(const char *data)
+{
+    if (HasNulByte(data, 10))
+    {
+        return -1;
+    }
+    if (!isdigit(data[0]))
+    {
+        return -1;
+    }
+    size_t length;
+    int ret = sscanf(data, "%10zd", &length);
```

Comment:
What is the point of reading ``size_t`` here and then converting it to ``long``?

Reply:
Also, this should use ``PREFIX_LENGTH``.

Reply:
The length is a size_t, the return value is a long (since it can be -1).

Reply:
I believe ``size_t`` can also be -1.